tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -4114,3 +4115,278 @@ def test_16_barbican(self):
+                        raise Exception("Create encrypted workload with attached ceph volume")
+                else:
+                    raise Exception("Create encrypted workload with attached ceph volume")
+
+                snapshot_id = self.workload_snapshot(wid, True, snapshot_cleanup=False)
+                self.wait_for_workload_tobe_available(wid)
+                snapshot_status = self.getSnapshotStatus(wid, snapshot_id)
+                if (snapshot_status == "available"):
+                    reporting.add_test_step("Create full snapshot", tvaultconf.PASS)
+                    mount_path = self.get_mountpoint_path(
+                        tvaultconf.tvault_ip[0], tvaultconf.tvault_username,
+                        tvaultconf.tvault_password)
+                    snapshot_found = self.check_snapshot_exist_on_backend(
+                        tvaultconf.tvault_ip[0], tvaultconf.tvault_username,
+                        tvaultconf.tvault_password, mount_path,
+                        wid, snapshot_id)
+                    LOG.debug(f"snapshot_found: {snapshot_found}")
+                    if snapshot_found:
+                        reporting.add_test_step("Verify snapshot existence on " \
+                                                "target backend", tvaultconf.PASS)
+                        self._check_encryption_on_backend(wid, snapshot_id,
+                                                          vm_id, disk_names, mount_path)
+                    else:
+                        raise Exception("Verify snapshot existence on target backend")
+                else:
+                    raise Exception("Create full snapshot")
+
+                vol_snap_name = "TrilioVaultSnapshot"
+
+                restore_id = ""
+                if (restore_test[1] == "selective"):
+                    trilio_vol_snapshots_before = self.get_trilio_volume_snapshot(vol_snap_name)
+                    # selective restore
+                    rest_details = {}
+                    rest_details['rest_type'] = 'selective'
+                    rest_details['network_id'] = CONF.network.internal_network_id
+                    rest_details['subnet_id'] = self.get_subnet_id(
+                        CONF.network.internal_network_id)
+                    rest_details['instances'] = {vm_id: volumes}
+                    payload = self.create_restore_json(rest_details)
+
+                    # Trigger selective restore of full snapshot
+                    restore_id = self.snapshot_selective_restore(
+                        wid, snapshot_id,
+                        restore_name="selective_restore_full_snap",
+                        instance_details=payload['instance_details'],
+                        network_details=payload['network_details'], restore_cleanup=False)
+                    self.wait_for_snapshot_tobe_available(wid, snapshot_id)
+
+                    if (self.getRestoreStatus(wid, snapshot_id, restore_id) == "available"):
```

Comment:
restore verification and clean-up is repeated for all restore types, can be converted to function and called for all restore types

Reply:
Verification is changing for all 3 restores, selective and oneclick does not change vol snapshot name during restore; however vol snapshots deletion is changed. 
In place restore creates vol snapshot during restore.